.github/workflows/miscellaneous.yml

```diff
@@ -26,3 +26,12 @@ jobs:
     # Ensure all .rs files have an Apache 2.0 SPDX license ID.
     - name: All .rs files have SPDX IDs
       run: .tests/misc-licenses-rs-spdx
+
+  misc-diagrams:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    # Ensure all diagrams are up-to-date.
+    - name: All diagrams are up-to-date
+      run: .tests/check-diagrams
```

Comment:
Rule of thumb we've settled on for tests: the filename should match the test name in the workflow. `check-diagrams` should be renamed to `misc-diagrams`.

If you anticipate that we'll add more diagram tests, you can name them more specifically (see how I subcategorized our licensing tests above this).